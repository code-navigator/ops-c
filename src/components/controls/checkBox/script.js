export default {
  name: 'checkBox',

  props: [
    'label',
    'value'
  ],

  computed: {
    input: {
      get () {
        return this.value
      },
      set (newValue) {
        this.$emit('keyup', newValue)
      }
    }
  }
}
