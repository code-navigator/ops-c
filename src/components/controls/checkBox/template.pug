//- Checkbox control
v-flex(xs12)
  v-checkbox(
    :label="label"
    v-model="input"
  )