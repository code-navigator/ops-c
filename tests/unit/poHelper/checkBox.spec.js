import { createLocalVue, mount } from '@vue/test-utils'
import { TestHelpers } from './../testHelpers'
import checkBox from './../../../src/components/controls/checkBox'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuetify)
localVue.use(Vuex)

describe('test.vue', () => {
  let h
  let store
  let wrapper
  const label = 'some label'
  const value = false

  beforeEach(() => {
    store = new Vuex.Store({})
    wrapper = mount(checkBox, {
      localVue,
      store,
      propsData: {
        label,
        value
      }
    })
    h = new TestHelpers(wrapper, expect)
    jest.resetModules()
    jest.clearAllMocks()
  })

  it('should mount without errors', () => {
    h.isInstance()
  })

  it('should render correctly', () => {
    h.renders()
  })

  it('should display a label', () => {
    h.hasText(label)
  })

  it('should reflect current state of value prop', async () => {
    h.hasDataPropWithValue('value', false)
    await h.click('input')
    h.hasDataPropWithValue('value', false)
  })
})
