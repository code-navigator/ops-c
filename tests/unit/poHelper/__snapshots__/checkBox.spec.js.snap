// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test.vue should render correctly 1`] = `
<div class="flex xs12">
  <div class="v-input v-input--selection-controls v-input--checkbox">
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-input--selection-controls__input">
          <input aria-label="some label" aria-checked="false" role="checkbox" type="checkbox">
          <div class="v-input--selection-controls__ripple"></div>
          <i aria-hidden="true" class="v-icon material-icons">check_box_outline_blank</i>
        </div>
        <label aria-hidden="true" class="v-label" style="left: 0px; position: relative;">some label</label>
      </div>
      <div class="v-messages">
        <div class="v-messages__wrapper"></div>
      </div>
    </div>
  </div>
</div>
`;
